sensor/common/heritage/heritage.go

```diff
@@ -0,0 +1,280 @@
+package heritage
+
+import (
+	"cmp"
+	"context"
+	"fmt"
+	"slices"
+	"sync/atomic"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/version"
+	v1 "k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	cmName             = "sensor-past-instances"
+	configMapKey       = "heritage"
+	annotationInfoKey  = `stackrox.io/past-sensors-info`
+	annotationInfoText = `This data is for sensor to recognize its past pod instances.`
+
+	// heritageMinSize is set to 2 as the smallest reasonable minimum covering 1 entry about the past and 1 about the
+	// current sensor. Setting this to 1 would disable the historical data and make the heritage feature useless.
+	heritageMinSize = 2
```

Comment:
I think this should be something like:
```suggestion
	// heritageMinSize is set to 1 as the smallest reasonable minimum covering 1 entry about the past.
	// Setting this to 0 disables the historical data.
	heritageMinSize = 1
```

Reply:
> So you suggest to set the minimum to 1 but then in the code always remember that it must be 1 past and 1 current?

This.
So value 0 will naturally disable the feature.

Reply:
Okay, I can do that, but setting that to 0 will not disable the feature - it will keep storing the data about the current instance of sensor and there will be always exactly one entry. This is not useful in any case, that is why I decided to set it explicitly to `2`.
Do you maybe want to revise you suggestion?

Reply:
Okay, I will, but still I find it important that we understand what this feature does and why I chose this number.

The case:
> 1: store one value: sensor reads the value on startup and overrides it. It stores both in operational memory, though, to distinguish live and dead sensors.

is valid but has no business value, as the data about currently running sensor is already known to the clusterEntitesStore and there is no need to remember it here additionally, because it won't "survive" the crash.
Assuming that a single entry is stored in the configMap, then after the crash of Sensor, that data would not be read but instead it will be immediately overwritten by new value. There is absolutely no guarantee that the reading from the configmap and storing in clusterEntitesStore (not part of this PR) will happen before new values are written to the configmap.

Reply:
> There is absolutely no guarantee that the reading from the configmap and storing in clusterEntitesStore (not part of this PR) will happen before new values are written to the configmap.
This is surprising. I cannot argue without seeing the whole picture, and am left in doubts.

I understood that the minimum value of 2 is there, actually, due to the random order of reading and writing to the ConfigMap.

I assumed that reading and updating of the configMap is a single synchronous procedure, that  happens before anything else to ensure the data is updated before a potential crash.

Reply:
I understand that the inner-workings of Sensor may be complex. Providing a guarantee that the informers will always start after the data is read from the configmap is possible but that would add complexity that we don't want to add without having a good reason for it. I don't think that changing the min value from 2 to 1 is sufficient to justify the additional complexity.

Regarding the reading from the configmap, there are two situations where the data is read:
1. (this PR) Reading in order to append new row to the slice and then write the slice back to the configmap. This must be synchronous and reading must happen before writing on each execution. That is also the reason for caching the values in the Manager, so that we only read from the configmap once and can write multiple times.
2. (not this PR) Reading in order to feed the data into the `clusterEntitiesStore`. This won't be in sync with the k8s informers and may happen any time as soon as the `clusterEntitiesStore` is created and k8s client is ready.

Reply:
So `clusterEntitiesStore` won't know if the data it has read from the `configMap` is updated or not, and therefore whether the last IP is the current or a heritage value.

Reply:
The `clusterEntitiesStore` doesn't need to know which value is the current and which is historical.